builder.ts

```diff
@@ -74,6 +75,12 @@ export class PackDefinitionBuilder implements BasicPackDefinition {
    * See {@link PackVersionDefinition.version}.
    */
   version?: string;
+
+  /**
+   * See {@link PackVersionDefinition.sdkVersion}.
+   */
+  sdkVersion?: string;
```

Comment:
Could we maybe do something like have `validatePackVersionMetadata()` take the sdkVersion as a second argument and inject it into into the `metadata` value before calling `safeParse()`? (Then remove it before returning it.)

I'd rather that the SDK version remain an out-of-band piece of data that the maker doesn't even have the opportunity to specify themselves, it should always be inferred by the tooling.

Reply:
Updated so `validatePackVersionMetadata()` takes an extra argument and consults a list called `packMetadataSchemaBySdkVersion` that can modify the Zod schema conditionally based on the SDK version. Seemed a little cleaner for the moment than mutating the `metadata` temporarily, but I don't have strong feelings either way